app/controllers/api/connect/v3/systems/systems_controller.rb

```diff
@@ -7,12 +7,9 @@ def update
       params[:online_at].each do |online_at|
         dthours = online_at.split(':')
         if dthours.count == 2
-          begin
-            @system_uptime = SystemUptime.create!(system_id: @system.id, online_at_day: dthours[0], online_at_hours: dthours[1])
-            logger.debug(N_("Added uptime information for system '%s'") % @system.id)
-          rescue ActiveRecord::RecordNotUnique
-            logger.debug(N_("Uptime information existing for system '%s'") % @system.id)
-          end
+          SystemUptime.create_or_update(system: @system,
```

Comment:
I would not use this method as it already exists in ActiveRecord.

I would propose the following change:

```
online_day = online_at.split(':')[0]
online_hours = online_at.split(':')[1]
system_uptime = @system.system_uptimes.find_by(online_at_day: online_day)

if system_uptime
  system_uptime.update!(online_at_hours: online_hours)
else
  SystemUptime.create!(system_id: @system.id, online_at_day: online_day, online_at_hours: online_hours)
end
```